notify/impl.go

```diff
@@ -62,6 +66,79 @@ func Build(confs []*config.Receiver, tmpl *template.Template) map[string]Fanout
 
 const contentTypeJSON = "application/json"
 
+// End-users should not be exposed to Go's type system,
```

Comment:
I agree that the `model.LabelName`/`model.LabelValue` types do not need to be exposed. Don't agree with the statement in general though, e.g. in Prometheus templating, I don't think dates and such should be plain strings.

Lots of infrastructure software is written in Go. Most of them (Docker, Kubernetes) allow accessing data by defining templates. Anyone making use of such features can be expected to be able to deal with the well-defined template system.